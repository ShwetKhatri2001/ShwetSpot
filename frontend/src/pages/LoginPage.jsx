import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { Button, Col, Form, Row } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import Message from '../components/Message';
import Loader from '../components/Loader';
import { getUserDetails, login } from '../redux/actions/userActions';

const LoginPage = ({ location, history }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const dispatch = useDispatch();

  const userLogin = useSelector(state => state.userLogin);
  const { loading, error, userInfo } = userLogin;

  const redirect = location.search ? location.search.split('=')[1] : '/';

  useEffect(() => {
    if (userInfo) {
      dispatch(getUserDetails('profile'));
      history.push(redirect);
    }
  }, [history, userInfo, redirect, dispatch]);

  const submitHandler = e => {
    e.preventDefault();
    dispatch(login(email, password));
  };

  return (
    <div className="justify-content-center align-items-center container bg-light p-4">
      <h1>Sign In</h1>
      {error && <Message variant="danger">{error}</Message>}
      {loading && <Loader />}
      <Form onSubmit={submitHandler}>
        <Form.Group controlId="email">
          <Form.Label>Email Address</Form.Label>
          <Form.Control
            type="email"
            placeholder="Enter email"
            value={email}
            onChange={e => setEmail(e.target.value)}
          />
        </Form.Group>

        <Form.Group controlId="password">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            placeholder="Enter password"
            value={password}
            onChange={e => setPassword(e.target.value)}
          />
        </Form.Group>

        <Button type="submit" variant="success">
          Sign In
        </Button>
      </Form>

      <Row className="py-3">
        <Col>
          Don't have an account?{' '}
          <Link to={redirect ? `/register?redirect=${redirect}` : '/register'}>
            Sign up
          </Link>
        </Col>
      </Row>
    </div>
  );
};

export default LoginPage;
