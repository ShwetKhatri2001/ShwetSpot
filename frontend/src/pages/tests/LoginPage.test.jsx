import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import LoginPage from '../LoginPage';
import store from '../../redux/store';

describe('LoginPage Component', () => {
  const MockLoginPage = (
    <MemoryRouter>
      <Provider store={store}>
        <LoginPage location={{ search: '1' }} />
      </Provider>
    </MemoryRouter>
  );

  it('should match the snapshot', () => {
    const tree = renderer.create(MockLoginPage).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
