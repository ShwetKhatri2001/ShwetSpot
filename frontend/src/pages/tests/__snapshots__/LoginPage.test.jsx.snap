// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Component should match the snapshot 1`] = `
<div
  className="justify-content-center align-items-center container bg-light p-4"
>
  <h1>
    Sign In
  </h1>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="email"
      >
        Email Address
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        placeholder="Enter email"
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
        value=""
      />
    </div>
    <button
      className="btn btn-success"
      disabled={false}
      type="submit"
    >
      Sign In
    </button>
  </form>
  <div
    className="py-3 row"
  >
    <div
      className="col"
    >
      Don't have an account?
       
      <a
        href="/register"
        onClick={[Function]}
      >
        Sign up
      </a>
    </div>
  </div>
</div>
`;
